src/dask_awkward/lib/optimize.py

```diff
@@ -234,14 +241,23 @@ def _touch_all_data(*args, **kwargs):
 
 def _mock_output(layer):
     """Update a layer to run the _touch_all_data."""
-    assert len(layer.dsk) == 1
+    if _dask_uses_tasks:
+        new_layer = copy.deepcopy(layer)
+        task = new_layer.task.copy()
```

Comment:
My guess is that the task specific copy is not required after the deepcopy. I was already contemplating whether we should get rid of `copy` (because it is difficult to maintain / would require subclasses to overwrite it and we might want to make use of subclassing)